lib/sinicum/imaging/imaging.rb

```diff
@@ -90,8 +94,17 @@ def perform_conversion
             convert(in_file.path, out_file.path)
             FileUtils.mv(out_file.path, file_rendered)
             FileUtils.chmod(0644, file_rendered)
+            out_index = out_file.path.rindex("-out-")
```

Comment:
This is way too complicated. The `Imaging` class should be responsible for **one** image, that you want to render. You can pass **one** srcset_option as parameter to determine the highres factor and the naming of the file, but the rest should work as before.